ui/desktop/electron-app/src/utils/sanitizer.js

```diff
@@ -0,0 +1,33 @@
+const { URL } = require('url');
+
+module.exports = {
+  /**
+   * Super paranoid shell quote/escape and validation.  Input must be base62.
+   * @param {string} str
+   */
+  base62EscapeAndValidate: (str) => {
+    const candidate = str.toString();
+    if (candidate.match(/^[A-Za-z0-9_]*$/)) return candidate;
+    throw new Error(`
+      Could not invoke command:
+      input contained unsafe characters.
+    `);
+  },
+  /**
+   * Return a quoted string suitable for using in shell commands.
+   * @param {string} str
+   * @returns {string}
+   */
+  urlValidate: (str) => {
+    try {
+      const url = new URL(str);
+      return url.href;
```

Comment:
Probably want to be a bit more intentional here.  `href` will return most of the URL:

```
> new URL('http://localhost:9200?foo=bars;test').href
'http://localhost:9200/?foo=bars;test'
```

I suggest reconstructing the origin: `${url.protocol}://${url.hostname}:${url.port}`

Reply:
Which raises the question, should this parsing and reconstitution happen within the runtime settings when origin is set in the first place?

Reply:
When origin is being updated, it will now be validated that it is a valid url in addition to verifying scopes api reachability. I left boundary command url validation as is for now. We can remove it in the future if it is found to be frivolous.